dirs/dirs.go

```diff
@@ -190,6 +198,73 @@ func init() {
 	SetRootDir(root)
 }
 
+// SnapHomeDirs returns a slice of the home directories currently in use.
+func SnapHomeDirs() []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+	dirs := make([]string, len(snapHomeDirs))
+	copy(dirs, snapHomeDirs)
+	// Should never be true since SetSnapHomeDirs is ran on init and on SetRootDir calls.
+	// Useful for unit tests.
+	if len(dirs) == 0 {
+		return []string{filepath.Join(GlobalRootDir, "/home")}
+	}
+	return dirs
+}
+
+// SetSnapHomeDirs sets SnapHomeDirs to the user defined values and appends /home if needed.
+// Homedir must be a comma separated list of the user defined home directories.
+// If homedirs is empty, SnapHomeDirs will be a slice of length 1 containing "/home".
+// Also generates the data directory globbing expressions for each user.
+// Ran at startup by configstate.Init, returns the split outsnap home directories. Use DataHomeGlobs to retrieve globs.
+func SetSnapHomeDirs(homedirs string) []string {
+	snapHomeDirsMu.Lock()
+	defer snapHomeDirsMu.Unlock()
+
+	//clear old values
+	snapHomeDirs = nil
+	snapDataHomeGlob = nil
+	hiddenSnapDataHomeGlob = nil
+
+	// Do not set the root directory as home unless explicitly specified with "."
+	if homedirs != "" {
+		snapHomeDirs = strings.Split(homedirs, ",")
+		for i := range snapHomeDirs {
+			// clean the path
+			snapHomeDirs[i] = filepath.Clean(snapHomeDirs[i])
+			globalRootDir := GlobalRootDir
+			// Avoid false positives with HasPrefix
```

Comment:
I find this comment confusing. I suspect the intent is so for the `globalRootDir` to have a trailing slash. The code, however could misbehave if the path was other than `/` and already had the trailing slash.

This would only affect tests but I think it should look more like:

``` go
if globalRootDir != "/" && !strings.HasSuffix(globalRootDir, "/") {
   globalRootDir += "/"
}
```

Reply:
Thank you for the review! 
With the current implementation the root directory cannot have a trailing '/' at the end due to `filepath.Clean(rootdir)` being called in `SetRootDir()`. The reason I called `filepath.Clean(rootdir)` in `SetRootDir()` was to avoid future checks such as `strings.HasSuffix(globalRootDir, "/")` that might also be needed in other parts of the source or will in the future. `filepath.Clean(rootdir)` also makes sure that the format of `rootdir` is predictable. 

However having said that I don't mind reverting it if still necessary.

Reply:
Note that `filepath.Clean` does not modify the argument so that separately looks suspicious. I think regardless of that constraint, code should be obvious and less clever to do the right thing.

Regardless of everything else `globalRootDir` may be something other than "/" in tests, so the code should be able to handle that.